### PR TITLE
Bag input feature tf2 porting

### PR DESCRIPTION
**Pull request by ydudin3**
# Code Pull Requests

Please provide the following:

- a clear explanation of what your code dose
- if applicable, a reference to an issue
- a reproducible test for your PR (code, model definition and data sample)

# Documentation Pull Requests

Note that the documentation HTML files are in `docs/` while the Markdown sources are in `mkdocs/docs`.

If you are proposing a modification to the documentation you should change only the Markdown files.

`api.md` is automatically generated from the docstrings in the code, so if you want to change something in that file, first modify `ludwig/api.py` docstring, then run `mkdocs/code_docs_autogen.py`, which will create `mkdocs/docs/api.md` .

### CODE REVIEW

File: ludwig/models/modules/bag_encoders.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            force_embedding_size=False,
+            embeddings_on_cpu=False,
+            dropout_rate=0.0,
+            initializer=None,
+            regularizer=None,
+            **kwargs
+    ):
+        super(BagEmbedWeightedEncoder, self).__init__()
+
+        self.embed_sparse = EmbedWeighted(
```
Comment (w4nderlust) [inline ludwig/models/modules/bag_encoders.py:44]: self.embed_sparse -> self.embed_weighted ?

File: ludwig/models/modules/embedding_modules.py [hunk 2]
```diff
@@ -163,6 +163,8 @@ def __init__(
     ):
         super(EmbedWeighted, self).__init__()
 
+        self.vocab = vocab
```
Comment (w4nderlust) [inline ludwig/models/modules/embedding_modules.py:166]: not super sure why you need this, is self.vocab used somewhere else?

**Comment by w4nderlust** [comment]
Left a couple comments, minor stuff.
Other than that, does this pass the tests? If yes I'll merge once the comments are addressed.

File: ludwig/models/modules/embedding_modules.py (hunk 2 above)
Comment (ydudin3) [reply ludwig/models/modules/embedding_modules.py:166]: It's referenced in line 203.

File: ludwig/models/modules/bag_encoders.py (hunk 1 above)
Comment (ydudin3) [reply ludwig/models/modules/bag_encoders.py:44]: Yep, fixed that.

**Comment by ydudin3** [comment]
@w4nderlust `test_experiment_various_feature_types` which uses bag feature as input passes locally.

File: ludwig/models/modules/embedding_modules.py (hunk 2 above)
Comment (w4nderlust) [reply ludwig/models/modules/embedding_modules.py:166]: You are right. But there we are only using the length, so maybe it's worth just doing:
```
self.vocab_length = len(vocab)
```
and then using it in line 203.

Comment (ydudin3) [reply ludwig/models/modules/embedding_modules.py]: Makes sense, updated.

**Pull request merged**